### ISSUE TITLE
parseMultiple is broken in multiple ways.

### ISSUE THREAD

**Issue by Jean-Daniel** [marmelroy/PhoneNumberKit#249]
parseMultiple creates a Serial Dispatch Queue, and uses DispatchGroup to perform an iteration which can't be faster than using a naive for loop, as it introduces Dispatch overhead without providing any benefit.

First, the canonical way to do work in parallel is to use `DispatchQueue.concurrentPerform`. Any other method is flawed as it can't properly manage the number of created thread and may cause serious issues.

Second, the executed block is in no way thread safe. Parsing the number require access to the RegexManager which is not thread safe due to the use of a cache to save parsed patterns.
Then, the appending of the result in multiParseArray isn't safe either.

Is only works today because the parsing is executed on a serial queue.

**Comment by wellhereismyusername** [comment]
True, `parseMultiple` doesn't do anything useful, only potentially dangerous. 
The reason it is now faster than calling `parse` in a loop is in the `PhoneNumberKit.defaultRegionCode()` call used as the default `region` argument in `parse`. It decreases the performance of repeated calls of `parse` more than 10 times

**Comment by Jean-Daniel** [comment]
It is still broken and poorly implemented.

**Comment by bguidolim** [comment]
I guess it was fixed with #529

Can you confirm it please?
By the way, we are open for PRs.

Please remember that I'm not able to publish new versions to Cocoapods, so you must set the git directly in your Podfile.

**Comment by Jean-Daniel** [comment]
It's still subject to thread explosion and is slow by design, due to the use of DispatchQueue and Dispatch group instead of using `DispatchQueue.concurrentPerform`

**Comment by bguidolim** [comment]
@Jean-Daniel @wellhereismyusername Could you guys check the PR #535
